workflows/mhcquant.nf

```diff
@@ -100,200 +33,139 @@ include { INPUT_CHECK } from '../subworkflows/local/input_check'
-    ch_versions = ch_versions.mix(OPENMS_PEPTIDEINDEXER.out.versions.ifEmpty(null))
+    OPENMS_PEPTIDEINDEXER(OPENMS_COMETADAPTER.out.idxml.combine(ch_decoy_db))
+    ch_versions = ch_versions.mix(OPENMS_PEPTIDEINDEXER.out.versions)
 
     // Save indexed runs for later use to keep meta-run information. Sort based on file id
     OPENMS_PEPTIDEINDEXER.out.idxml
-            .map { meta, idxml -> [[id: meta.sample + '_' + meta.condition], meta] }
-            .groupTuple( sort: sortById )
-            .set { merge_meta_map }
+        .map { meta, idxml -> [ groupKey([id: "${meta.sample}_${meta.condition}"], meta.group_count), meta] }
+        .groupTuple()
+        .set { merge_meta_map }
 
     OPENMS_PEPTIDEINDEXER.out.idxml
-            .map { meta, idxml -> [[id: meta.sample + '_' + meta.condition], idxml] }
-            .groupTuple()
-            .set { ch_runs_to_merge }
+        .map { meta, idxml -> [ groupKey([id: "${meta.sample}_${meta.condition}"], meta.group_count), idxml] }
+        .groupTuple()
+        .set { ch_runs_to_merge }
 
     // Merge aligned idXMLfiles
     OPENMS_IDMERGER(ch_runs_to_merge)
-    ch_versions = ch_versions.mix(OPENMS_IDMERGER.out.versions.ifEmpty(null))
-
-    // Extract PSM features for Percolator
-    OPENMS_PSMFEATUREEXTRACTOR(OPENMS_IDMERGER.out.idxml)
-    ch_versions = ch_versions.mix(OPENMS_PSMFEATUREEXTRACTOR.out.versions.ifEmpty(null))
-
-    // Run Percolator
-    OPENMS_PERCOLATORADAPTER(OPENMS_PSMFEATUREEXTRACTOR.out.idxml)
-    ch_versions = ch_versions.mix(OPENMS_PERCOLATORADAPTER.out.versions.ifEmpty(null))
-
-    // Filter by percolator q-value
-    OPENMS_IDFILTER_Q_VALUE(OPENMS_PERCOLATORADAPTER.out.idxml.flatMap { it -> [tuple(it[0], it[1], null)] })
-    ch_versions = ch_versions.mix(OPENMS_IDFILTER_Q_VALUE.out.versions.ifEmpty(null))
+    ch_versions = ch_versions.mix(OPENMS_IDMERGER.out.versions)
 
-    //
-    // SUBWORKFLOW: Refine the FDR values on the predicted subset
-    //
-    if (params.refine_fdr_on_predicted_subset && params.predict_class_1) {
-        // Run the following subworkflow
-        REFINE_FDR (
-            OPENMS_IDFILTER_Q_VALUE.out.idxml,
-            OPENMS_PSMFEATUREEXTRACTOR.out.idxml,
-            peptides_class_1_alleles
-        )
-        ch_versions = ch_versions.mix(REFINE_FDR.out.versions.ifEmpty(null))
-        // Define the outcome of the paramer to a fixed variable
-        filter_q_value = REFINE_FDR.out.filter_refined_q_value
+    // Run MS2Rescore
+    ch_clean_mzml_file
+            .map { meta, mzml -> [ groupKey([id: "${meta.sample}_${meta.condition}"], meta.group_count), mzml] }
+            .groupTuple()
+            .join(OPENMS_IDMERGER.out.idxml)
+            .map { meta, mzml, idxml -> [meta, idxml, mzml, []] }
+            .set { ch_ms2rescore_in }
+
+    MS2RESCORE(ch_ms2rescore_in)
+    ch_versions = ch_versions.mix(MS2RESCORE.out.versions)
+
+    if (params.rescoring_engine == 'percolator') {
```

Comment:
This if-and-else statement can also be put into the module.config right?

Reply:
I don't think so since I need to call a different combination of modules in the workflow depending on that parameter 🤔